bpf/lib/nodeport.h

```diff
@@ -3346,14 +3357,20 @@ lb_handle_health(struct __ctx_buff *ctx __maybe_unused, __be16 proto)
 }
 #endif /* ENABLE_HEALTH_CHECK */
 
+/* handle_nat_fwd() handles revDNAT, fib_lookup_redirect, and bpf_snat for
+ * nodeport. If revdnat_only is set to 1, fib_lookup and bpf_snat are skipped.
+ * The typical use case of handle_nat_fwd(revdnat_only=1) is for handling reply
+ * traffic that requires revDNAT prior to wireguard/IPsec encryption.
+ */
 static __always_inline int
 handle_nat_fwd(struct __ctx_buff *ctx, __u32 cluster_id, __be16 proto,
-	       struct trace_ctx *trace __maybe_unused,
+	       __u32 revdnat_only, struct trace_ctx *trace __maybe_unused,
 	       __s8 *ext_err __maybe_unused)
 {
 	int ret = CTX_ACT_OK;
 
 	ctx_store_meta(ctx, CB_CLUSTER_ID_EGRESS, cluster_id);
+	ctx_store_meta(ctx, CB_REVDNAT_ONLY, revdnat_only);
 
```

Comment:
To expand on Martynas' comment - my experience is that we shouldn't pass booleans through their own CB slots (the result looks like [this](https://github.com/cilium/cilium/blob/29029ddca1daba9354f5b42384777466ebf86a45/bpf/lib/l3.h#L107) - using three slots for three booleans).

So can we maybe turn this into a `CB_NAT_FLAGS` instead, and pass a `#define NAT_FLAGS_REVDNAT_ONLY (1 << 0)` ? Then this hopefully becomes much easier to extend later.